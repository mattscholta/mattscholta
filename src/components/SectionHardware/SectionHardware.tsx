import * as React from 'react';
import classnames from 'classnames';

import * as links from '@src/data/uses';

import styles from './styles.module.css';

export interface SectionHardwareProps {
  className?: string;
}

/**
 * @name SectionHardware
 * @description Stateless Component / Pure Function
 */
const SectionHardware: React.FC<SectionHardwareProps> = (props) => {
  const { className } = props;

  // Styles
  const cssHardware = classnames('ui-list-disc u-flex-grow ', styles.hardware);

  // Setup
  const propsLink = { rel: 'noopener noreferrer', target: '_blank' };

  return (
    <section className={className}>
      <h2 className="ui-heading">Hardware</h2>
      <sub className="ui-subhead" />

      <p>
        I've been happily developing on a <b>Mac</b> over the last <b>10+</b>{' '}
        years. However, I grew up on PC's and built a few rigs over the years
        for gaming and work. Now days a perk of working in Software our employer
        often supplies a well equipped box.
      </p>

      <div className="u-flex">
        <ul className={cssHardware}>
          <li>
            Laptop -{' '}
            <a {...propsLink} href="https://support.apple.com/kb/SP776">
              15" MacBook Pro, 2018
            </a>
          </li>
          <li>
            Monitor -{' '}
            <a
              {...propsLink}
              href="https://www.amazon.com/gp/product/B074JKT894"
            >
              34" LG Curved Ultra-wide
            </a>
          </li>
          <li>
            Storage -{' '}
            <a {...propsLink} {...propsLink} href={links.hrefEnclosure}>
              TerraMaster RAID Enclosure
            </a>
          </li>
          <li className="nested">
            <a {...propsLink} {...propsLink} href={links.hrefHardDrives}>
              2x WD Gold 4TB @7200 RPM
            </a>
          </li>

          <li>
            Desk -{' '}
            <a
              {...propsLink}
              href="https://www.amazon.com/gp/product/B07HJJLLML"
            >
              Adjustable height Desk
            </a>
          </li>
          <li>
            Keyboard -{' '}
            <a href="https://www.apple.com/shop/product/MLA22B/A/magic-keyboard-us-english">
              Apple Magic Keyboard
            </a>
          </li>
          <li>
            Music -{' '}
            <a
              {...propsLink}
              href="https://www.amazon.com/Bose-QuietComfort-Wireless-Headphones-Cancelling/dp/B0756CYWWD"
            >
              Bose Headphones
            </a>
          </li>
          <li>
            Mouse -{' '}
            <a
              {...propsLink}
              href="https://www.apple.com/shop/product/MLA02LL/A/magic-mouse-2-silver"
            >
              Apple Magic Mouse
            </a>
          </li>
        </ul>
      </div>
    </section>
  );
};

export { SectionHardware };
