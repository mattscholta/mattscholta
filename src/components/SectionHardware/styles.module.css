.hardware {
  column-count: 2;
}

@media (max-width: 768px) {
  .hardware {
    column-count: 1;
  }
}
