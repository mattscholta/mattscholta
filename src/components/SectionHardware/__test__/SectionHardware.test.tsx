import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { SectionHardware, SectionHardwareProps } from '../SectionHardware';

describe('SectionHardware', () => {
  let props: SectionHardwareProps;
  let wrapper: RenderResult;

  const createWrapper = () => <SectionHardware {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
