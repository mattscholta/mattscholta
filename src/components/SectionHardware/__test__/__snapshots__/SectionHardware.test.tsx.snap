// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHardware matches snapshot 1`] = `
NodeList [
  <section>
    <h2
      class="ui-heading"
    >
      Hardware
    </h2>
    <sub
      class="ui-subhead"
    />
    <p>
      I've been happily developing on a 
      <b>
        Mac
      </b>
       over the last 
      <b>
        10+
      </b>
       
      years. However, I grew up on PC's and built a few rigs over the years for gaming and work. Now days a perk of working in Software our employer often supplies a well equipped box.
    </p>
    <div
      class="u-flex"
    >
      <ul
        class="list-disc u-flex-grow  hardware"
      >
        <li>
          Laptop -
           
          <a
            href="https://support.apple.com/kb/SP776"
            rel="noopener noreferrer"
            target="_blank"
          >
            15" MacBook Pro, 2018
          </a>
        </li>
        <li>
          Monitor -
           
          <a
            href="https://www.amazon.com/gp/product/B074JKT894"
            rel="noopener noreferrer"
            target="_blank"
          >
            34" LG Curved Ultra-wide
          </a>
        </li>
        <li>
          Storage -
           
          <a
            href="https://www.amazon.com/gp/product/B01MRSRQLA"
            rel="noopener noreferrer"
            target="_blank"
          >
            TerraMaster RAID Enclosure
          </a>
        </li>
        <li
          class="nested"
        >
          <a
            href="https://www.amazon.com/gp/product/B01AV1697A"
            rel="noopener noreferrer"
            target="_blank"
          >
            2x WD Gold 4TB @7200 RPM
          </a>
        </li>
        <li>
          Desk -
           
          <a
            href="https://www.amazon.com/gp/product/B07HJJLLML"
            rel="noopener noreferrer"
            target="_blank"
          >
            Adjustable height Desk
          </a>
        </li>
        <li>
          Keyboard -
           
          <a
            href="https://www.apple.com/shop/product/MLA22B/A/magic-keyboard-us-english"
          >
            Apple Magic Keyboard
          </a>
        </li>
        <li>
          Music -
           
          <a
            href="https://www.amazon.com/Bose-QuietComfort-Wireless-Headphones-Cancelling/dp/B0756CYWWD"
            rel="noopener noreferrer"
            target="_blank"
          >
            Bose Headphones
          </a>
        </li>
        <li>
          Mouse -
           
          <a
            href="https://www.apple.com/shop/product/MLA02LL/A/magic-mouse-2-silver"
            rel="noopener noreferrer"
            target="_blank"
          >
            Apple Magic Mouse
          </a>
        </li>
      </ul>
    </div>
  </section>,
]
`;
